crmd/te_events.c

```diff
@@ -565,23 +581,33 @@ process_graph_event(xmlNode * event, const char *event_node)
         desc = "arrived late";
         abort_transition(INFINITY, tg_restart, "Inactive graph", event);
 
-    } else if (match_graph_event(action, event, event_node, status, rc, target_rc) < 0) {
-        desc = "unknown";
-        abort_transition(INFINITY, tg_restart, "Unknown event", event);
+    } else {
+        action = get_action(action_num, FALSE);
 
-    } else if (rc == target_rc) {
-        passed = TRUE;
-        crm_trace("Processed update to %s: %s", id, magic);
+        if (action == NULL) {
+            desc = "unknown";
+            abort_transition(INFINITY, tg_restart, "Unknown event", event);
+
+        } else {
+            /* XML_ATTR_TE_ALLOWFAIL will be true if on-fail=ignore for the operation */
+            ignore_failures = crm_is_true(crm_meta_value(action->params,
+                                                         XML_ATTR_TE_ALLOWFAIL));
```

Comment:
I probably would have done this inside `match_graph_event()` since it already has `action` passed into it, but thats just a style issue